### DISCUSSION TITLE
Trouble updating fortls

### DISCUSSION THREAD

**Discussion by kayleighember** [fortran-lang/fortls#128] [Q&A] [answered]
Hi there, sorry if this is a really dumb question but how do you restart the server? VS Code keeps trying to update fortls for me, but it says I need to restart it. I've tried ending the task, restarting VS Code, and restarting Windows.
![image](https://user-images.githubusercontent.com/58703205/170061153-25ab8d2c-ab34-47af-8f3e-5990933d196b.png)

Also, VS Code seems to think I'm on fortls version 2.5.0, while powershell thinks I have 2.6.0. What am I doing wrong here?

**Comment by gnikit** [comment, marked as answer]
One way of restarting is to close and relaunch VS Code. The new way is pressing `Ctrl`+`Shift`+`P` to bring up the command pallet. if you start typing `Fortran` an option should pop up that says, `Fortran: Restart the Fortran Language Server`.

As for the difference in version, I suspect that it might be because VS Code has not restarted yet. Another explanation could be that you have installed fortls in 2 different locations, you can [get the location](https://superuser.com/questions/675837/equivalent-of-cmds-where-in-powershell) of fortls and check if it is the same that is being used in VS Code.

FYI the autoupdate is from `fortls` itself not VS Code. Make sure you have an internet connection and allow for the install process to complete. You can check the progress of pip install in the log channel (see below)

![image](https://user-images.githubusercontent.com/16143716/170076202-311bdfab-7ac9-4039-860f-ad3804a1315f.png)

**Reply by kayleighember** [reply]
Thanks so much! Yeah, when I did a `pip show fortls` in the integrated terminal, it showed me that I had 2.5.0 installed in `C:\Users\username\AppData\Roaming\Python\Python310\site-packages`. I uninstalled it from the integrated terminal, and now it seems to be getting 2.6.0 from `C:\Python310\Scripts` just fine!

**Reply by gnikit** [reply]
I'm glad I was able to help @kayleighember, thanks for reaching out.
